### PR TITLE
Changes to support more distros in Sysbox

### PR DESCRIPTION
**Pull request by ctalledo**

### CODE REVIEW

File: rootfsCloner/rootfsCloner.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+package rootfsCloner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/nestybox/sysbox-libs/formatter"
+	"github.com/opencontainers/runc/libcontainer/mount"
```
Comment (rodnymolina) [inline rootfsCloner/rootfsCloner.go:26]: Why don't you use sysbox-runc's equivalent package instead? Unless we're using a recent change from oci-runc (not present in sysbox-runc yet), there shouldn't be any runc dependency from sysbox-mgr (or any other sysbox component) -- the exception being sysbox-runc itself, of course.

File: rootfsCloner/utils.go [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+func filePerm(path string) (os.FileMode, error) {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return os.FileMode(0), err
+	}
+	return fi.Mode().Perm(), nil
+}
+
+// Creates the cloned rootfs overlayfs mounts and bind-mounts over the orig rootfs.
+func mountClone(ci *cloneInfo) error {
```
Comment (rodnymolina) [inline rootfsCloner/utils.go:53]: Neat. Thinking about how to re-use this pattern for other/new features (extra security) ...

**Review by rodnymolina — COMMENTED** [review]
Some comments below. Need some more time for deeper review of `rootfsclone` logic.

File: main.go [hunk 3]
```diff
@@ -94,9 +94,13 @@ func main() {
 			Value: "/var/lib/sysbox",
 			Usage: "root directory for sysbox data store",
 		},
-		cli.BoolTFlag{
-			Name:  "bind-mount-id-shift",
-			Usage: "When set to true, Sysbox performs user and group ID shifting (e.g., shiftfs mounts) on host directories bind-mounted to the system containers so that they appear with correct ownership inside the container (default = true)",
+		cli.BoolFlag{
+			Name:  "disable-shiftfs",
+			Usage: "Disables Sysbox's use of the kernel's shiftfs module; files may show with nobody:nogroup ownership inside the container meant for testing. (default = false)",
```
[the remark this answers is not in the document]
Comment (guobao-BG) [reply main.go:99]: I want add --device option, could you give some informations?

[the remark this answers is not in the document]
Comment (ctalledo) [reply main.go:99]: Hi @guobao-BG, do you mean exposing host hardware devices onto the sysbox container? (e.g., `docker run --sysbox-runc --device ..`). Or do mean something else?

File: rootfsCloner/rootfsCloner.go (hunk 1 above)
Comment (ctalledo) [reply rootfsCloner/rootfsCloner.go:26]: Good point, fixed.

**Review by rodnymolina — APPROVED** [review]
LG

**Pull request merged**